hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -1582,6 +1582,10 @@ void startStandbyServices(final Configuration conf, boolean isObserver)
       standbyCheckpointer = new StandbyCheckpointer(conf, this);
       standbyCheckpointer.start();
     }
+    if (isNoManualAndResourceLowSafeMode()) {
+      LOG.info("Standby should not enter safe mode when resources are low, exiting safe mode.");
+      leaveSafeMode(false);
```

Comment:
It is reasonable at first glance, not think carefully, any cases to trigger Standby leave safemode untimely? Thanks.

Reply:
I reused the logic from [HDFS-17231](https://issues.apache.org/jira/browse/HDFS-17231), and I believe there is no issue. HDFS-17231 enables the ANN to automatically exit ResourceLowSafeMode. 
At the same time, I noticed that the 'leaveSafeMode(false)' method also exits 'StartupSafeMode'. I'm not sure if this is an issue; I mentioned this phenomenon in [HDFS-17402](https://issues.apache.org/jira/browse/HDFS-17402).
If necessary, I can fix it.

Reply:
The mention in "HDFS-2915" is that SNN should not enter resource low safe mode. NNRM thread removed from SNN.

Reply:
Consider the following scenario:

Initially, nn-1 is active and nn-2 is standby. The insufficient resources of both nn-1 and nn-2 in dfs.namenode.name.dir, the NameNodeResourceMonitor detects the resource issue and puts nn01 into safemode.
At this point, nn-1 is in safemode (ON) and active, while nn-2 is in safemode (OFF) and standby.
After a period of time, the resources in nn-2's dfs.namenode.name.dir recover, triggering failover.
Now, nn-1 is in safe mode (ON) and standby, while nn-2 is in safe mode (OFF) and active.
Afterward, the resources in nn-1's dfs.namenode.name.dir recover.
However, since nn-1 is standby but in safemode (ON), it unable to exit safe mode automatically.

Reply:
Great. Got it. Please check if the failed unit test is related with this changes. Others look good to me.

Reply:
Thanks for your review. I passed the failed unit test locally.